### PR TITLE
[21] [22] [23] As a logged-in user, I can select repositories to notify

### LINKED ISSUE #21: [UI] As a logged-in user, I can select repositories to notify
## Why

To support the logged-in user to select repositories they want for notifications, we need to create a repositories preferences screen

## Acceptance Criteria

- When the user presses the `Notify` button on each repository, we will update it to the `Notifying` button (follow the design)
- When the user presses the `Notifying` button on each repository, we will update it to the `Notify` button (follow the design)

Design: https://www.figma.com/file/jCSHtb5pycYvT57B5MZaSu/Github-Notifications?node-id=7%3A108

### LINKED ISSUE #22: [Backend] As a logged-in user, I can select repositories to notify
## Why

To support the logged-in user to select repositories they want for notifications, we need to create a repositories preferences screen

## Acceptance Criteria

- When the user presses on the `Notify` button:
    - We need to save the repository to the UserDefault, then later, we only filter the selected repository for notifications.
- When the user presses the `Notifying` button:
    - We remove the selected repository from the UserDefault, then later, we cannot receive any notifications from that repository.

Design: https://www.figma.com/file/jCSHtb5pycYvT57B5MZaSu/Github-Notifications?node-id=7%3A108

### LINKED ISSUE #23: [Integrate] As a logged-in user, I can select repositories to notify
## Why
To support the logged-in user to select repositories they want for notifications, we need to create a repositories preferences screen

## Acceptance Criteria
- When the user presses on the Notify button:
    - We need to save the repository to the UserDefault, then later, we only filter the selected repository for notifications.
- When the user presses the Notifying button:
    - We remove the selected repository from the UserDefault, then later, we cannot receive any notifications from that repository.

Design: https://figma.com/file/jCSHtb5pycYvT57B5MZaSu/Github-Notifications?node-id=7%3A108

### PR DESCRIPTION
**Pull request by llleyelll**
close #21
close #22
close #23

## What happened 👀

- Store notified repositories to `AppStorage`.
- Integrate select repositories to notify.
 
## Insight 📝

- #22 is quite small, so I decided to work along with integration.
- #21 is implemented in #56
- Add and remove a repository from the notified list when tapping on the button.
- Add extension for `Array` to allow `AppStorage`  to store `Array` type.
- Update `ContentView` sidebar.
 
## Proof Of Work 📹

Kapture 2021-09-16 at 12 24 16

### CODE REVIEW

File: Source/Application/Constants/UserStorage.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+//
+//  Created by Chananchida F. on 9/15/21.
+//  Copyright © 2021 Nimblehq. All rights reserved.
+//
+
+import Foundation
+
+enum UserStorage {
+
+    @UserDefault(key: .selectedRepos, defaultValue: [String]())
```
Comment (blyscuit) [inline Source/Application/Constants/UserStorage.swift:13]: please take a look at [AppStorage](https://www.hackingwithswift.com/quick-start/swiftui/what-is-the-appstorage-property-wrapper) also example in `GeneralScreenObservable`

File: Source/Application/Constants/UserStorage.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//  UserStorage.swift
+//  GithubNotifications
+//
+//  Created by Chananchida F. on 9/15/21.
+//  Copyright © 2021 Nimblehq. All rights reserved.
+//
+
+import Foundation
+
+enum UserStorage {
```
Comment (blyscuit) [inline Source/Application/Constants/UserStorage.swift:11]: You will use this in a Singleton way, I suggest creating a protocol that will help with unit test. Also I suggest making this an object with `get` and `set` functions.

File: Source/Application/Constants/UserStorage.swift (hunk 1 above)
Comment (llleyelll) [reply Source/Application/Constants/UserStorage.swift]: @blyscuit The storing notified repositories part is quite small, so I included the integration in this PR as well. Please help check 🙏

File: Source/Views/RepositoryCell/RepositoryCell.swift [hunk 3]
```diff
@@ -11,7 +11,9 @@ import SwiftUI
 struct RepositoryCell: View {
 
     @State var viewModel: RepositoryCellViewModel
-    @State var isNotifying: Bool = true
+    @State var isNotifying: Bool
+
+    @ObservedObject var notifiedReposObservable = NotifiedRepositoryObservable()
```
Comment (blyscuit) [inline Source/Views/RepositoryCell/RepositoryCell.swift:16]: This works fine but to improve this, current we have two separated instances of `NotifiedRepositoryObservable`, I think we can use `@Binding` for `isNotifying` which is being passthrough from `SearchRepoScreen`'s `notifiedReposObservable`. You can take a look at this for fun (and make it more Combine-ish)

**Review by blyscuit — APPROVED** [review]

File: Source/Module/Main/ContentView.swift [hunk 4]
```diff
@@ -14,13 +14,14 @@ struct ContentView: View {
 
     init() {
         _isLoggedIn = State(initialValue: viewModel.getIsLoggedIn())
+        UITableViewCell.appearance().selectionStyle = .none
```
Comment (markgravity) [inline Source/Module/Main/ContentView.swift:17]: I think we can do it by using `List.listStyle()` 🤔

Comment (llleyelll) [reply Source/Module/Main/ContentView.swift:17]: @markgravity I set `listStyle` to be `.listStyle(SidebarListStyle())` but the highlight on tap still appear. I also tried this [approach](https://www.hackingwithswift.com/quick-start/swiftui/how-to-disable-the-overlay-color-for-images-inside-button-and-navigationlink) but it doesn't work for some reason. 🥲
```Swift
NavigationLink(
    ...
)
.buttonStyle(PlainButtonStyle())
```

**Review by suho — APPROVED** [review]

**Pull request merged**